hpx/parallel/algorithms/for_each.hpp

```diff
@@ -223,22 +226,15 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
             sequential(ExPolicy, InIter first, InIter last, F && f,
                 Proj && proj)
             {
+                typedef typename util::detail::loop<InIter>::type it_type;
+
                 return util::loop(first, last,
                     [&f, &proj](Iter curr)
                     {
                         f(hpx::util::invoke(proj, *curr));
                     });
```

Comment:
Test fails because of this line, that:

'/hpx/hpx/parallel/algorithms/for_each.hpp:231:24: error: no viable conversion from returned value of type 'hpx::local_raw_vector_iterator<int, __gnu_cxx::__normal_iterator<int *, std::vector<int, std::allocator<int> > > >' to function return type 'hpx::local_vector_iterator<int>'

In all lambdas in this file, ' hpx::util::invoke(f, hpx::util::invoke(proj, *curr)); ' is used except this line, shouldn't we change 'f(hpx::util::invoke(proj, *curr)); '  to ' hpx::util::invoke(f, hpx::util::invoke(proj, *curr)); '  as well?